### PR TITLE
feat: APPS-2707 Create FTVA BlockInfo component

### PR DESCRIPTION
**Pull request by tinuola**
Connected to [APPS-2707](https://jira.library.ucla.edu/browse/APPS-2707)

**Component Created:** BlockInfo.vue
**Stories:** ~/stories/BlockInfo.stories.js
**Spec:** ~/stories/BlockInfo.spec.js

**Storybook Preview:** https://deploy-preview-529--ucla-library-storybook-vue3x.netlify.app/?path=/docs/block-info--docs

Notes: Additional work done
- Refactored ButtonLink component
  - Changed setup to composition API
  - Moved styles into modular/theme-based setup 
  - Created FTVA button stories

**Component Modified:** ButtonLink.vue
**Stories:** ~/stories/ButtonLink.stories.vue

**Storybook Preview:** https://deploy-preview-529--ucla-library-storybook-vue3x.netlify.app/?path=/docs/button-link--docs

---

**Screenshot:**

BlockInfo

**Checklist:**

-   [x] I checked that it is working locally in the dev server
-   [x] I checked that it is working locally in the storybook
-   ~~[ ] I checked that it is working locally in the 
library-website-nuxt dev server~~
-   [x] I added a screenshot of it working
-   [x] UX has reviewed and approved this
-   [x] I have requested a PR review from the dev team
-   [x] I used a conventional commit message
-   [x] I assigned myself to this PR

[APPS-2707]: https://uclalibrary.atlassian.net/browse/APPS-2707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by farosFreed — CHANGES_REQUESTED** [review]
Made some comments

File: src/lib-components/BlockInfo.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  ftvaTicketInformation: {
+    type: Array,
+    default: () => [],
+  },
+
+  // Note / ToDo:
+  /* This prop is for the potential edge case that an event page will require
+    a registration link.
+
+    Although this field exists in Craft, the design spec is pending.
```
[the remark this answers is not in the document]
Comment (tinuola) [reply src/lib-components/BlockInfo.vue:17]: Yes, as far as I understand.

File: src/styles/ftva/_block-info.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.block-info {
```
Comment (farosFreed) [inline src/styles/ftva/_block-info.scss:1]: needs to be `.ftva.block-info` or theme will have no effect

**Review by farosFreed — APPROVED** [review]
looks good to me

**Pull request merged**